### PR TITLE
[JS Polyfills] Number.toLocaleString() polyfill

### LINKED ISSUE #988: Number.toLocaleString() polyfill
Seems like .toLocaleString is not supported (ie, formatting thousand separators correctly depending on user locale, for instance showing 1000 as 1,000 in the US and as 1 000 in Sweden.

### PR DESCRIPTION
**Pull request by pewallin**
Fixes #988 for my personal use case. Really simple polyfill that formats thousand separators for the default or choosen locale. Does not support the more advanced uses of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString (such as currency or other options).

### CODE REVIEW

**Comment by brentvatne** [comment]
Hmm, I'm starting to wonder if this might be better suited as an external lib. @vjeux what are your thoughts on adding in additional polyfills to the react-native core both in general and also specifically in this case? :smile:

File: Libraries/JavaScriptAppEngine/System/JSNumber/JSToLocaleString.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            no: {
+              delimiter: ',',
+              decimal: '.'
+            }
+          },
+          locale,
+          navigatorLanguage,
+          localeCharacters;
+
+        navigatorLanguage = (navigator && navigator.language) || 'en';
```
Comment (vjeux) [inline Libraries/JavaScriptAppEngine/System/JSNumber/JSToLocaleString.js]: navigator.language is undefined on react native, which makes this entire polyfill mostly useless.

**Comment by vjeux** [comment]
I think that react-native environment should behave as much as possible like state of the art browser and follow the specs.

In this case, this is not a great polyfill because it doesn't follow the specifications at all. It doesn't work in languages other than english and only supports one option. A bad polyfill is worse than no polyfill as it's likely going to cause a lot of problems down the line.

Now, the thing that sucks about polyfills is that you have to load them all the time and it costs startup time for things that are probably not going to be used by a lot of people. I'd love to get to a point where you can npm install to-locale-string-polyfill and now you have it.

cc @sebmarkbage who put a lot of thoughts on the subject

**Comment by brentvatne** [comment]
That makes a lot of sense @vjeux - thanks for the detailed response, we can refer to it in the future for other polyfill discussions

**Pull request closed without merging**